### ISSUE TITLE
save weights for more than last 5 epoch

### ISSUE THREAD

**Issue by Xiaoyang-Rebecca** [JiahuiYu/generative_inpainting#288]
Hi Jiahui,
I would like to appreciate your excellent work of inpainting. I am trying to use it on the recovery of the compressed images.

I have a few questions about training weights:

1)  In the default .yml file, the GAN_LOSS_ALPHA is set to 0.001.
https://github.com/JiahuiYu/generative_inpainting/blob/06cd62cfca8c10c349b451fa33d9cbb786bfaa20/inpaint.yml#L10
 Is it a typo and supposed to be 0.0005 for wgan?  Does it mean you have a fixed step size value rather than decreased fixed value?

2) What is the loss value printed out in the logger?
https://github.com/JiahuiYu/neuralgym/blob/ab6603651236ab2485c2821dae1b2a71f6d75d28/neuralgym/train/trainer.py#L140
 Is it for discriminator only? Is it for **training loss** or **validation loss**?

3) If it is training loss, could you help to explain the increasing loss like this? 
![image](https://user-images.githubusercontent.com/29383759/60845557-55563e00-a1a2-11e9-8257-2120c4b4ff89.png)

Although from https://github.com/JiahuiYu/generative_inpainting/issues/53, you mentioned it is normal to have small turbulence of training loss, but I am assuming it shouldn't keep increasing. 
How do you set a wiser way of stop training **except for manually stop training** or **manually set MAX_ITERS**?

4) If there is no way to stop training automatically, what if I would like to exact the optimal weights manually (e.g. when the loss is the minimum). Then how could we save the weights for **all epochs**.
 In the model_log folder, it only showed the lastest 5 snap- of the weights, (i.e. the previous weights has removed) 
https://github.com/JiahuiYu/generative_inpainting/blob/06cd62cfca8c10c349b451fa33d9cbb786bfaa20/train.py#L108

**Comment by JiahuiYu** [comment]
@Xiaoyang-Rebecca Hi, thanks for your interest in our work first.

I also appreciate your clean organization of questions.

1. It is not a typo. We slightly tuned that hyper-parameter and that one works slightly better. But you are free to do another hyper-parameter sweep.

2. Training loss.

3. The training loss may increase because you are jointly optimizing generator and discriminator. For more reasons, please search keywords like GAN, DCGAN, WGAN, etc.

4. Usually you don't need to save all weights. Because the network does not seem to overfit on training images. But for your question, that's tensorflow-related. I guess there is some flags in tensorflow that can meet your requirement.

**Comment by Xiaoyang-Rebecca** [comment]
@JiahuiYu Thank you very much for your response!
